gazebo_plugins/test/test_gazebo_ros_wheelslip.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gazebo/test/ServerFixture.hh>
+#include <gazebo_ros/node.hpp>
+#include <gazebo_ros/testing_utils.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+using namespace std::chrono_literals;
+
+/// Tests the gazebo_ros_wheelslip plugin
+class GazeboRosWheelSlipTest : public gazebo::ServerFixture
+{
+public:
+  void TestParameters(
+    std::string world_file,
+    std::string node_name,
+    std::map<std::string, double> parameter_pairs)
+  {
+    // Load the world
+    this->Load(world_file, true);
+    auto world = gazebo::physics::get_world();
+    ASSERT_NE(nullptr, world);
+    auto node = std::make_shared<rclcpp::Node>("test_gazebo_ros_wheelslip");
+    ASSERT_NE(nullptr, node);
+
+    // Create parameters client
+    auto parameters_client = std::make_shared<rclcpp::SyncParametersClient>(
+      node, node_name);
+
+    // Wait for serive to show up
+    int counter = 0;
+    while (!parameters_client->wait_for_service(1s)) {
+      std::cout << "service not available, waiting again..." << std::endl;
+      counter++;
+      if (counter > 5) {FAIL();}
+    }
+
+    // Verify the parameters
+    std::vector<std::string> parameter_names;
+    for (auto & element : parameter_pairs) {
+      parameter_names.push_back(element.first);
+    }
+    std::this_thread::sleep_for(1s);
+    auto parameters_received = parameters_client->get_parameters(parameter_names);
+    for (auto & parameter : parameters_received) {
+      ASSERT_EQ(parameter_pairs[parameter.get_name()], parameter.as_double());
+    }
+  }
+};
+
+
+TEST_F(GazeboRosWheelSlipTest, TestWorldFile_2)
```

Comment:
Looking at the implementation of each test, it's difficult to know _what_ we are testing. The comment is good, but it might be more obvious if we renamed the tests to something more informative, e.g. `TestRosParametersOverrideSdf`.

Reply:
Renamed test cases